test/server/cards/01_SOR/VigilantHonorGuards.spec.js

```diff
@@ -0,0 +1,45 @@
+describe('Vigilant Honor Guards', function() {
+    integration(function() {
+        describe('Vigilant Honor Guards\' ability', function() {
+            beforeEach(function () {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['consular-security-force']
+                    },
+                    player2: {
+                        groundArena: ['vigilant-honor-guards'],
+                        hand: ['repair']
+                    }
+                });
+            });
+
+            it('should give it sentinel only as long as it is undamaged', function () {
+                this.player1.clickCard(this.consularSecurityForce);
+                //Honor Guards automatically selected due to sentinel
+                expect(this.player2).toBeActivePlayer();
+                expect(this.vigilantHonorGuards.damage).toBe(3);
+                expect(this.consularSecurityForce.damage).toBe(4);
+
+                this.moveToNextActionPhase();
+
+                this.player1.clickCard(this.consularSecurityForce);
+                expect(this.player1).toBeAbleToSelectExactly([this.vigilantHonorGuards, this.p2Base]); //no sentinel
+                this.player1.clickCard(this.p2Base);
+
+                //checking if they regain sentinel when fully healed
+                this.player2.clickCard(this.repair);
+                this.player2.clickCard(this.vigilantHonorGuards);
+                expect(this.vigilantHonorGuards.damage).toBe(0);
+
+                this.moveToNextActionPhase();
```

Comment:
Prefer to just un-exhaust the unit when that's an option since this will move through the entire regroup phase, which might be kind of slow. Probably not noticeable for a single test but it'll add up